kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolverTest.java

```diff
@@ -149,13 +152,35 @@ void resolveProxyRefsWithResolvedRefsFalseCondition() {
         assertThat(resolutionResult.allClustersInNameOrder()).containsExactly(cluster);
         ClusterResolutionResult onlyResult = assertSingleResult(resolutionResult, cluster);
         assertThat(onlyResult.allReferentsFullyResolved()).isTrue();
-        assertThat(ResourcesUtil.hasResolvedRefsFalseCondition(onlyResult.cluster())).isTrue();
+        assertThat(ResourcesUtil.hasFreshResolvedRefsFalseCondition(onlyResult.cluster())).isTrue();
+    }
+
+    @Test
+    void resolveProxyRefsWithStaleResolvedRefsFalseCondition() {
```

Comment:
I think the test name should be improved.  How about something like:
shouldNotResolveProxyRefsWhenGenerationIsStale